controllers/oauth.controller.js

```diff
@@ -0,0 +1,171 @@
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcryptjs");
+const nodemailer = require('nodemailer');
+const Oauth = require("../models/oauth.schema");
+const User = require("../models/user.schema");
+const Otp = require("../models/otp.schema");
+const { postOauthSchema, patchUserPasswordSchema } = require("../validation/oauth.validation");
+const { postOtpSchema, patchOtpSchema } = require('../validation/otp.validation');
+const logger = require("../config/logger");
+require('dotenv').config();
+
+const userLogin = async (req, res) => {
+  try {
+    const { error, value } = postOauthSchema.validate(req.body);
+    if (error) {
+      return res.status(400).send({ message: error.message });
+    }
+    const user = await User.findOne(
+      { email: req.body.email },
+    );
+    value.password = await bcrypt.compare(value.password, user.password);
+
+    if (value.password === true) {
+      const userToken = await jwt.sign({ _id: user.id.toString() }, process.env.TOKEN_KEY, { expiresIn: "1h" });
+      const oauth = new Oauth({ token: userToken });
```

Comment:
You should move this in Mongoose Model's Statics - https://mongoosejs.com/docs/2.7.x/docs/methods-statics.html

And use it here this way:
```
 const { token } = await Oauth.generateNewToken(user.id)
 res.status(200).send({ token });
```